homeassistant/components/python_script/__init__.py

```diff
@@ -216,17 +226,32 @@ def protected_getattr(obj, name, default=None):
         "hass": hass,
         "data": data or {},
         "logger": logger,
+        "output": {},
     }
 
     try:
         _LOGGER.info("Executing %s: %s", filename, data)
         # pylint: disable-next=exec-used
         exec(compiled.code, restricted_globals)  # noqa: S102
+        _LOGGER.debug(
+            "Output of python_script: `%s`:\n%s",
+            filename,
+            restricted_globals["output"],
+        )
+        # Ensure that we're always returning a dictionary
+        if not isinstance(restricted_globals["output"], dict):
+            output_type = type(restricted_globals["output"])
+            restricted_globals["output"] = {}
+            raise ScriptError(
```

Comment:
Raising this error here is captured a couple of lines below.
If any output generating service causes an exception, it should be passed on.
See [this](https://github.com/home-assistant/core/pull/96695#discussion_r1265668347) comment.

Reply:
Took me a minute to understand what you're implying, but I get it now I think.

You're suggesting to re-raise the exceptions, right? So not just logging them, but actually stopping the execution.

Reply:
Correct. I'm not quite sure why myself, but that is a requirement for ServiceResponses